.github/workflows/sonarcloud.yml

```diff
@@ -0,0 +1,47 @@
+name: SonarCloud
+on:
+  push:
+    branches:
+      - '*'
+  pull_request:
+    branches:
+      - '*'
+
+jobs:
+  sonarcloud:
+    runs-on: ubuntu-latest
+    if: (github.event_name == 'push') || ((github.event_name == 'pull_request') && (github.event.pull_request.head.repo.full_name != github.repository))
```

Comment:
The first two conditionals are redundant I think?

Reply:
they are needed to prevent duplicate runs from triggering if a new update is made to a PR